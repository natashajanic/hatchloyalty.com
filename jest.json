{
  "collectCoverage": true,
  "coveragePathIgnorePatterns": ["tests"],
  "globals": {},
  "moduleFileExtensions": [
    "ts",
    "tsx",
    "js"
  ],
  "moduleNameMapper": {
    "^.+\\.(css|less|scss)$": "identity-obj-proxy",
    "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/tests/assetsTransformer.ts",
    "^src/(.*)$": "<rootDir>/src/$1"
  },
  "setupFiles": [
    "raf/polyfill"
  ],
  "setupFilesAfterEnv": [
    "<rootDir>/tests/setupTests.ts"
  ],
  "snapshotSerializers": [
    "enzyme-to-json/serializer"
  ],
  "transform": {
    "\\.(ts|tsx)$": "ts-jest"
  },
  "testRegex": "/src/.*/__tests__/.*\\.(ts|tsx|js)$",
  "unmockedModulePathPatterns": [],
  "verbose": true
}
