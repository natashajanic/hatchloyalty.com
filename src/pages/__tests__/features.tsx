import { shallow, ShallowWrapper } from 'enzyme'
import 'jest' // tslint:disable-line no-import-side-effect
import * as React from 'react'
import FeaturesPage from '../features'

describe('FeaturesPage', () => {
  let mountedComponent: ShallowWrapper<any, any, FeaturesPage>

  beforeEach(() => {
    mountedComponent = shallow(<FeaturesPage />)
  })

  describe('render', () => {
    it("renders six sections that cover the platform's features", async () => {
      expect(mountedComponent.find('section').length).toEqual(6)
    })

    it('matches the expected snapshot', async () => {
      expect(mountedComponent).toMatchSnapshot()
    })
  })
})
