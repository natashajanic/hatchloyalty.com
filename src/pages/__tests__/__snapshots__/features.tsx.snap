// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturesPage render matches the expected snapshot 1`] = `
<Layout>
  <SEO
    keywords={
      Array [
        "hatch",
        "hatch loyalty",
        "loyalty",
        "personalization",
        "activation",
        "features",
      ]
    }
    title="Features"
  />
  <Styled(Component)
    blacklist={
      Array [
        "css",
        "maxWidth",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
      ]
    }
    is="main"
    m="0 auto"
    maxWidth="1024px"
    px={
      Array [
        3,
        4,
      ]
    }
    py={5}
  >
    <section>
      <Styled(Component)
        alignItems="center"
        blacklist={
          Array [
            "css",
            "alignItems",
            "border",
            "borderColor",
            "borderBottom",
            "borderRadius",
            "color",
            "bg",
            "display",
            "justifyContent",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
            "width",
          ]
        }
        display="flex"
      >
        <Styled(Component)
          bg="green"
          blacklist={
            Array [
              "css",
              "borderRadius",
              "display",
              "m",
              "mt",
              "mr",
              "mb",
              "ml",
              "mx",
              "my",
              "p",
              "pt",
              "pr",
              "pb",
              "pl",
              "px",
              "py",
              "alignItems",
              "border",
              "borderColor",
              "borderBottom",
              "color",
              "bg",
              "justifyContent",
              "width",
            ]
          }
          borderRadius={100}
          display="inline-flex"
          p={2}
        >
          <Layers
            color="#fff"
            size={20}
          />
        </Styled(Component)>
        <Text
          blacklist={
            Array [
              "css",
              "fontSize",
              "lineHeight",
              "m",
              "mt",
              "mr",
              "mb",
              "ml",
              "mx",
              "my",
              "p",
              "pt",
              "pr",
              "pb",
              "pl",
              "px",
              "py",
              "textStyle",
              "alignItems",
              "color",
              "bg",
              "display",
              "flex",
              "fontWeight",
              "textAlign",
            ]
          }
          fontSize={1}
          lineHeight="1.5"
          m={0}
          ml={2}
          textStyle="normal"
        >
          Hatch Insights
        </Text>
      </Styled(Component)>
      <h2>
        A modern platform designed to make your loyalty-driven life infinitely easier.
      </h2>
      <h3>
        Hatch offers a comprehensive suite of flexible, dynamic, intuitive products that help you better know & serve your customers, and better manage & grow your business.
      </h3>
    </section>
    <hr />
    <section>
      <Styled(Component)
        alignItems="center"
        blacklist={
          Array [
            "css",
            "alignItems",
            "border",
            "borderColor",
            "borderBottom",
            "borderRadius",
            "color",
            "bg",
            "display",
            "justifyContent",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
            "width",
          ]
        }
        display="flex"
      >
        <Styled(Component)
          bg="green"
          blacklist={
            Array [
              "css",
              "borderRadius",
              "display",
              "m",
              "mt",
              "mr",
              "mb",
              "ml",
              "mx",
              "my",
              "p",
              "pt",
              "pr",
              "pb",
              "pl",
              "px",
              "py",
              "alignItems",
              "border",
              "borderColor",
              "borderBottom",
              "color",
              "bg",
              "justifyContent",
              "width",
            ]
          }
          borderRadius={100}
          display="inline-flex"
          p={2}
        >
          <UserCheck
            color="#fff"
            size={20}
          />
        </Styled(Component)>
        <Text
          blacklist={
            Array [
              "css",
              "fontSize",
              "lineHeight",
              "m",
              "mt",
              "mr",
              "mb",
              "ml",
              "mx",
              "my",
              "p",
              "pt",
              "pr",
              "pb",
              "pl",
              "px",
              "py",
              "textStyle",
              "alignItems",
              "color",
              "bg",
              "display",
              "flex",
              "fontWeight",
              "textAlign",
            ]
          }
          fontSize={1}
          lineHeight="1.5"
          m={0}
          ml={2}
          textStyle="normal"
        >
          Profiles
        </Text>
      </Styled(Component)>
      <h3>
        Get to know your customers beyond simple transactions to understand who they are and activate when it matters most.
      </h3>
      <h4>
        Hatch offers a comprehensive suite of flexible, dynamic, intuitive products that help you better know & serve your customers, and better manage & grow your business.
      </h4>
      <ul>
        <li>
          Pump
        </li>
        <li>
          Point of Sale
        </li>
        <li>
          Data Warehouse
        </li>
        <li>
          Marketing Automation
        </li>
        <li>
          Mobile App
        </li>
        <li>
          Website
        </li>
      </ul>
    </section>
    <hr />
    <section>
      <Styled(Component)
        alignItems="center"
        blacklist={
          Array [
            "css",
            "alignItems",
            "border",
            "borderColor",
            "borderBottom",
            "borderRadius",
            "color",
            "bg",
            "display",
            "justifyContent",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
            "width",
          ]
        }
        display="flex"
      >
        <Styled(Component)
          bg="green"
          blacklist={
            Array [
              "css",
              "borderRadius",
              "display",
              "m",
              "mt",
              "mr",
              "mb",
              "ml",
              "mx",
              "my",
              "p",
              "pt",
              "pr",
              "pb",
              "pl",
              "px",
              "py",
              "alignItems",
              "border",
              "borderColor",
              "borderBottom",
              "color",
              "bg",
              "justifyContent",
              "width",
            ]
          }
          borderRadius={100}
          display="inline-flex"
          p={2}
        >
          <BarChart
            color="#fff"
            size={20}
          />
        </Styled(Component)>
        <Text
          blacklist={
            Array [
              "css",
              "fontSize",
              "lineHeight",
              "m",
              "mt",
              "mr",
              "mb",
              "ml",
              "mx",
              "my",
              "p",
              "pt",
              "pr",
              "pb",
              "pl",
              "px",
              "py",
              "textStyle",
              "alignItems",
              "color",
              "bg",
              "display",
              "flex",
              "fontWeight",
              "textAlign",
            ]
          }
          fontSize={1}
          lineHeight="1.5"
          m={0}
          ml={2}
          textStyle="normal"
        >
          Wisdom
        </Text>
      </Styled(Component)>
      <h3>
        Get smarter, faster, with real-time data analysis and easily accessible, actionable insights.
      </h3>
      <h4>
        Wisdom turns complex data into easy-to-interpret information, allowing you to better manage your loyalty program from insight to execution. Activate offers in real-time with exceptional ease. Leverage data based on customers personal preferences. Monitor and create rewards on the fly.
      </h4>
    </section>
    <hr />
    <section>
      <Styled(Component)
        alignItems="center"
        blacklist={
          Array [
            "css",
            "alignItems",
            "border",
            "borderColor",
            "borderBottom",
            "borderRadius",
            "color",
            "bg",
            "display",
            "justifyContent",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
            "width",
          ]
        }
        display="flex"
      >
        <Styled(Component)
          bg="green"
          blacklist={
            Array [
              "css",
              "borderRadius",
              "display",
              "m",
              "mt",
              "mr",
              "mb",
              "ml",
              "mx",
              "my",
              "p",
              "pt",
              "pr",
              "pb",
              "pl",
              "px",
              "py",
              "alignItems",
              "border",
              "borderColor",
              "borderBottom",
              "color",
              "bg",
              "justifyContent",
              "width",
            ]
          }
          borderRadius={100}
          display="inline-flex"
          p={2}
        >
          <Activity
            color="#fff"
            size={20}
          />
        </Styled(Component)>
        <Text
          blacklist={
            Array [
              "css",
              "fontSize",
              "lineHeight",
              "m",
              "mt",
              "mr",
              "mb",
              "ml",
              "mx",
              "my",
              "p",
              "pt",
              "pr",
              "pb",
              "pl",
              "px",
              "py",
              "textStyle",
              "alignItems",
              "color",
              "bg",
              "display",
              "flex",
              "fontWeight",
              "textAlign",
            ]
          }
          fontSize={1}
          lineHeight="1.5"
          m={0}
          ml={2}
          textStyle="normal"
        >
          Moments
        </Text>
      </Styled(Component)>
      <h3>
        Impact purchase decisions right in the moment with personalized offers and rewards based on specific customer preferences.
      </h3>
      <h4>
        Moments, offers and rewards becomes truly personal and seriously effective. Create, activate, and modify offers in real time. Leverage data gathered through Profiles to provide personalized rewards and experiences that keep customers engaged and loyal.
      </h4>
    </section>
    <hr />
    <section>
      <Styled(Component)
        alignItems="center"
        blacklist={
          Array [
            "css",
            "alignItems",
            "border",
            "borderColor",
            "borderBottom",
            "borderRadius",
            "color",
            "bg",
            "display",
            "justifyContent",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
            "width",
          ]
        }
        display="flex"
      >
        <Styled(Component)
          bg="green"
          blacklist={
            Array [
              "css",
              "borderRadius",
              "display",
              "m",
              "mt",
              "mr",
              "mb",
              "ml",
              "mx",
              "my",
              "p",
              "pt",
              "pr",
              "pb",
              "pl",
              "px",
              "py",
              "alignItems",
              "border",
              "borderColor",
              "borderBottom",
              "color",
              "bg",
              "justifyContent",
              "width",
            ]
          }
          borderRadius={100}
          display="inline-flex"
          p={2}
        >
          <Repeat
            color="#fff"
            size={20}
          />
        </Styled(Component)>
        <Text
          blacklist={
            Array [
              "css",
              "fontSize",
              "lineHeight",
              "m",
              "mt",
              "mr",
              "mb",
              "ml",
              "mx",
              "my",
              "p",
              "pt",
              "pr",
              "pb",
              "pl",
              "px",
              "py",
              "textStyle",
              "alignItems",
              "color",
              "bg",
              "display",
              "flex",
              "fontWeight",
              "textAlign",
            ]
          }
          fontSize={1}
          lineHeight="1.5"
          m={0}
          ml={2}
          textStyle="normal"
        >
          Habits
        </Text>
      </Styled(Component)>
      <h3>
        Get to know your customers beyond simple transactions to understand who they are and activate when it matters most.
      </h3>
      <h4>
        Hatch offers a comprehensive suite of flexible, dynamic, intuitive products that help you better know & serve your customers, and better manage & grow your business.
      </h4>
      <h5>
        Segment By:
      </h5>
      <ul>
        <li>
          Member Activity
        </li>
        <li>
          Behavior
        </li>
        <li>
          Location
        </li>
        <li>
          Items Purchased
        </li>
        <li>
          Offers & Rewards
        </li>
        <li>
          Vists
        </li>
      </ul>
    </section>
    <hr />
    <section>
      <h5>
        Solutions
      </h5>
      <h3>
        Flexibility that meets a variety of needs, Hatch provides value for:
      </h3>
      <ul>
        <li>
          <h4>
            Brands & CPG's
          </h4>
          <p>
            Similar to retailers, you want to get to know you customers. Unlike retailers, you may not readily have access to rich customer data. Hatch enables brands & CPGs to access actionable customer data through our retail partners.
          </p>
        </li>
        <li>
          <h4>
            Agencies & Partners
          </h4>
          <p>
            If you’re an agency or tech provider who has a client that is looking for loyalty, but don’t want to deal with the cost and headache of building the tech yourself, Hatch’s flexible platform can integrate with your solution and initiatives.
          </p>
        </li>
        <li>
          <h4>
            Enterprise Retailers
          </h4>
          <p>
            The Hatch platform enables you to configure and design a program branded for your business. Most of what you’ll need is already built, but Hatch can enable unique experiences for your customers through integrations and extensions.
          </p>
        </li>
      </ul>
    </section>
  </Styled(Component)>
</Layout>
`;
